templates(netlify): improve DX for Netlify serverless template (#3754)
remix-run/remix: templates/netlify/README.md

## Development

> The Remix dev server starts your app in development mode, rebuilding assets on file changes.

Reviewer: [sand] Minor addition to make the connection between `npm run dev` and `remix dev` clearer.
```suggestion
The Remix dev server starts your app in development mode, rebuilding assets on file changes. To start the Remix dev server:
```

> The Remix dev server starts your app in development mode, rebuilding assets on file changes.

Reviewer: [pebble] It may make sense to sequence the content in this section. Do users need to have the Remix App Server running before starting `netlify dev`?

Even if they don't, as currently written, we say the Remix server starts the app in dev mode. Then we way the Netlify CLI does the same thing. And we don't really define the relationship, so it could be interpreted as either or both running the app in a non-specific way.

```suggestion
To run your Remix app locally and properly test Netlify features, you need to:
1. Start the Remix App Server in development mode
2. Start Netlify Dev

### Start the Remix App Server in development mode

The Remix dev server starts your app in development mode, rebuilding assets on file changes. To start the Remix dev server:
```

Author: No in this case, when you run `netlify dev`, it will build the code to run the Remix app server in Netlify functions.

```sh
npm run dev
```

Open up [http://localhost:3000](http://localhost:3000), and you should be ready to go!

> The Netlify CLI starts your app in development mode for testing the serverless server and any custom Netlify functions you've developed.

Reviewer: [pebble] Continuation of sequencing suggestion from above.

```suggestion
### Start Netlify Dev

The Netlify CLI starts your app in development mode for testing the serverless server and any custom Netlify functions you've developed.
```

> The Netlify CLI starts your app in development mode for testing the serverless server and any custom Netlify functions you've developed.

Reviewer: [pebble] Suggested alternative wording to avoid "serverless server".

```suggestion
The Netlify CLI builds a production version of your Remix App Server and splits it into Netlify Functions that run locally. This includes any custom Netlify functions you've developed. The Netlify CLI runs all of this in its development mode.
```

```sh
netlify dev
```

Open up [http://localhost:3000](http://localhost:3000), and you should be ready to go!

> Note: When running the Netlify CLI, file changes will rebuild assets, but you will not see the changes to the page you are on unless you do a browser refresh of the page. The reason being is the serverless version of the server does not support hot module reloading.

Reviewer 2: The wording here is a bit strange: "the serverless version of the server"!

> Note: When running the Netlify CLI, file changes will rebuild assets, but you will not see the changes to the page you are on unless you do a browser refresh of the page. The reason being is the serverless version of the server does not support hot module reloading.

Reviewer: [pebble] Suggested alternative wording to avoid "serverless server".
```suggestion
Note: When running the Netlify CLI, file changes will rebuild assets, but you will not see the changes to the page you are on unless you do a browser refresh of the page. Due to how the Netlify CLI builds the Remix App Server, it does not support hot module reloading.
```
